akka-stream/src/main/scala/akka/stream/scaladsl/SourceWithContext.scala

```diff
@@ -49,7 +48,11 @@ final class SourceWithContext[+Ctx, +Out, +Mat](
   def toMat[Mat2, Mat3](sink: Graph[SinkShape[(Out, Ctx)], Mat2])(combine: (Mat, Mat2) ⇒ Mat3): RunnableGraph[Mat3] =
     underlying.toMat(sink)(combine)
 
-  override def endContextPropagation: Prov[Ctx, Out] = underlying
+  /**
+   * Stops automatic context propagation from here and converts this to a regular
+   * stream of a pair of (data, context).
+   */
+  def endContextPropagation: Source[(Out, Ctx), Mat] = underlying
```

Comment:
Also rename the `startContextPropagation` to `withContext`? Patrik suggested the start.. end... methods, for consistency.

Reply:
I kept the name here intentionally. The reason for having different names between `SourceWithContext` and `FlowWithContext` is that they are used differently.

`SourceWithContext` can only be created from a `Source` so there should usually be a pair of `startWithContext` / `endWithContext`.

`FlowWithContext` on the other hand can be created on its own, so `endContextPropagation` sounds a bit weird.

Reply:
Ah, ok. I think I prefer `startWithContext` and `endWithContext` over `startContextPropagation` and `endContextPropagation`, just because of the length of the names.

Reply:
`startContextPropagation` feels more readable on one hand but on the other hand  `startWithContext` is technically also interesting because it aligns with `SourceWithContext`.

Reply:
yeah.. I'll leave it up to the naming gurus ;-)

Reply:
Though I think my preference is `startWithContext` and `endWithContext`